rpc/src/module/chain.rs

```diff
@@ -1645,4 +1692,107 @@ impl ChainRpc for ChainRpcImpl {
         );
         Ok(Some(median_time.into()))
     }
+
+    fn get_median_fee_rate(
+        &self,
+        blocks_to_scan: Option<Uint32>,
+        min_transactions: Option<Uint32>,
+    ) -> Result<ckb_jsonrpc_types::Capacity> {
+        let blocks_to_scan = blocks_to_scan
+            .map(|v| v.value())
+            .unwrap_or(MAX_BLOCKS_TO_SCAN);
+        let min_transactions = min_transactions
+            .map(|v| v.value())
+            .unwrap_or(MIN_TRANSACTIONS);
+
+        // check input parameter
+        if blocks_to_scan > MAX_BLOCKS_TO_SCAN {
+            return Err(RPCError::invalid_params(format!(
+                "invalid input parameter: max_blocks_to_scan, should be less than {}",
+                MAX_BLOCKS_TO_SCAN
+            )));
+        }
+        if min_transactions < MIN_TRANSACTIONS {
+            return Err(RPCError::invalid_params(format!(
+                "invalid input parameter: min_transactions, should be greater than {}",
+                MIN_TRANSACTIONS
+            )));
+        }
+
+        // find matched blocks
+        let snapshot = self.shared.snapshot();
+        let tip = snapshot.tip_header().number();
+        let mut match_blocks = Vec::with_capacity(blocks_to_scan as usize);
+        let mut counter = 0;
+
+        // TODO: parallel?
+        for index in (0..tip).rev() {
+            let block_hash = snapshot.get_block_hash(index).ok_or_else(|| {
+                RPCError::custom(RPCError::ChainIndexIsInconsistent, "get_block_hash error")
+            })?;
+            let block = snapshot.get_block(&block_hash).ok_or_else(|| {
+                RPCError::custom(RPCError::ChainIndexIsInconsistent, "get_block error")
+            })?;
+            // info!("transaction number:{} at {}", block.transactions().len(), index);
+            if block.transactions().len() >= min_transactions as usize {
+                match_blocks.push(block_hash);
```

Comment:
The RPC should collect non-cellbase transactions from recent blocks until:

- There are already `min_transactions` transactions
- Already scanned `blocks_to_scan` blocks


For example, if the current tip number is 10, and block 10 has 3 non-cellbase transactions, 9 has 2, 8 hash 3:

- When `blocks_to_scan` is 2 and `min_transactions` is 3, use transactions in block 10 to calculate the median.
- When `blocks_to_scan` is 2 and `min_transactions` is 4, use transactions in block 10 and 9 to calculate the median.
- When `blocks_to_scan` is 2 and `min_transactions` is 6, the RPC fails.

Reply:
yeah, i do finally realize these parameter really mean, when i read your comment...:-)